packages/frontend/src/lib/notification/GPUPromotion.spec.ts

```diff
@@ -0,0 +1,110 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import '@testing-library/jest-dom/vitest';
+import { beforeEach, expect, test, vi } from 'vitest';
+import { render, screen } from '@testing-library/svelte';
+import { studioClient } from '/@/utils/client';
+import type { ExtensionConfiguration } from '@shared/src/models/IExtensionConfiguration';
+import GPUPromotion from '/@/lib/notification/GPUPromotion.svelte';
+
+vi.mock('/@/utils/client', async () => {
+  return {
+    studioClient: {
+      navigateToResources: vi.fn(),
+    },
+  };
+});
+
+const mocks = vi.hoisted(() => {
+  return {
+    getConfigurationMock: vi.fn<() => ExtensionConfiguration>(),
+  };
+});
+
+vi.mock('../../stores/extensionConfiguration', () => ({
+  configuration: {
+    subscribe: (f: (msg: ExtensionConfiguration) => void) => {
+      f(mocks.getConfigurationMock());
+      return (): void => {};
+    },
+  },
+}));
```

Comment:
you can mock more easily the store 
https://github.com/containers/podman-desktop-extension-ai-lab/pull/1849#discussion_r1795172109